source/Engine.cpp

```diff
@@ -2032,11 +2032,11 @@ void Engine::HandleMouseClicks()
 // Determines alternate mouse turning, setting player mouse angle, and right-click firing weapons.
 void Engine::HandleMouseInput(Command &activeCommands)
 {
-	if(activeCommands.Has(Command::MOUSE_TURNING))
-	{
-		isMouseTurningEnabled = !isMouseTurningEnabled;
-		Preferences::Set("alt-mouse turning", isMouseTurningEnabled);
-	}
+	isMouseHoldEnabled = activeCommands.Has(Command::MOUSE_TURNING_HOLD);
+	if(activeCommands.Has(Command::MOUSE_TURNING_TOGGLE))
+		isMouseToggleEnabled = !isMouseToggleEnabled;
```

Comment:
```suggestion
	isMouseToggleEnabled = activeCommands.Has(Command::MOUSE_TURNING_TOGGLE) ?
		!isMouseToggleEnabled : isMouseToggleEnabled;
```
I think this can be compressed

Reply:
Does that actually result in less operations? Not sure if ternary operators are quicker, or just a convenience.

Reply:
The compiler almost certainly produces the same output for each of these (assuming optimisations are on, which they should be).
If it's very good, it'll convert it to a branchless form.
Instead of:
```c++
if(x)
    y = !y;
```
It'll do:
```c++
y = x ^ y
```

(This uses an XOR operation because it is equivalent.)

x | initial y | new y (current code) | x XOR initial y
-- | -- | -- | --
0 | 0 | 0 | 0
0 | 1 | 1 | 1
1 | 0 | 1 | 1
1 | 1 | 0 | 0

I prefer the current way with the full if statement because I think it is easiest to read.

Reply:
ok...